esx_service/vmdk_ops.py

```diff
@@ -668,20 +670,27 @@ def vm_uuid2name(vm_uuid):
         return None
     return vm.config.name
 
-# Return error, or None for OK.
-def attachVMDK(vmdk_path, vm_uuid):
-    vm = findVmByUuid(vm_uuid)
-    logging.info("*** attachVMDK: %s to %s VM uuid = %s",
-                 vmdk_path, vm.config.name, vm_uuid)
-    return disk_attach(vmdk_path, vm)
 
+def attach_or_detach_VMDK(cmd, vmdk_path, vm_name, bios_uuid, vc_uuid):
```

Comment:
Ideally a function should do one thing and only one thing. So I'd suggest to keep attach and detach as two separate functions. Meanwhile, we can extract the common logic to a new function, say findVm(...), to avoid code duplication.